server/src/lib/lazy-cache.ts

```diff
@@ -53,29 +76,13 @@ function redisCache<T, S>(
       const cached = JSON.parse(result);
       value = cached.value;
       renewCache = isExpired(cached.at, timeMs);
-    }
 
-    if (!renewCache) return value;
+      if (renewCache) renewRedisCache(key, f, timeMs)(...args);
+      return value;
+    }
 
     return new Promise(async resolve => {
-      const lock = await redlock.lock(
-        key + '-lock',
-        1000 * 60 * 3 /*3 minutes*/
-      );
-      const result = await redis.get(key);
-      if (result) {
-        const cached = JSON.parse(result);
-        renewCache = isExpired(cached.at, timeMs);
-        resolve(cached.value);
-      }
-
-      if (renewCache) {
-        value = await f(...args);
-        await redis.set(key, JSON.stringify({ at: Date.now(), value }));
-        resolve(value);
-      }
-
-      await lock.unlock();
+      resolve(await renewRedisCache(key, f, timeMs)(...args));
```

Comment:
I'll need to dive into the surrounding files to get more context on how / where this is being called, but I wonder why we're calling `renewRedisCache` in two places now?

Reply:
So the old logic used to be "if cached value exists and is not expired, return cached value. Under all other circumstances, call a fresh query and cache it in Redis, and when you're done, return that new value". Only one call to update the cached value was needed because whether the value expired or never existed in the first place, you had to wait for it to come back before doing anything. 

With the new logic, if the cached value exists but has expired, we're no longer waiting for the update function to come back because we're sending back the expired value regardless, but the call to update the expired value has to happen before the `return`. So we can't combine it with the same `await renew` that we'd need to perform if the value didn't exist at all. 

Does that make sense? It's totally possible I'm overcomplicating it.